odoo/fields.py

```diff
@@ -919,10 +920,40 @@ def update_db(self, model, columns):
 
         # create/update the column, not null constraint, indexes
         self.update_db_column(model, column)
+
+        if (not column
+            and self.related_field
+            and self.related_field.store
+            and len(self.related) == 2
+            and model._fields[self.related[0]].type not in {'one2many', 'many2many'}
+            and self.type not in {'one2many', 'many2many'}
+        ):
```

Comment:
A few remarks:
* `self` cannot be an x2many field because it is a column type (see condition above)
* `self.related_field and len(self.related) == 2` can be simplified as `len(self.related or ()) == 2`
* the intermediate field must be many2one, stored and not computed
* `self.related_field` must also be stored and not computed